### ISSUE TITLE
[9.0] Product variants are not available in Inventory (prev. Warehouse) view

### ISSUE THREAD

**Issue by aisopuro** [odoo/odoo#12406]
Impacted versions: 9.0 mainly, but I would extend this to 8.0 as well. Tested on `demo.odoo.com` just now.
#### Steps to reproduce:
1. In the settings, select "Products can have multiple attributes"
2. Open the Sales view.
3. In the Sales view, open the Sales dropdown: you will see menu options for both Products and Product Variants: 

![image](https://cloud.githubusercontent.com/assets/2587901/16035391/96b20618-321f-11e6-8191-fb50b7121229.png)
4. Go to the Inventory view
5. Open the Inventory control menu
#### Current behavior:

You will see an option for accessing the Products list, but not Product variants.

![image](https://cloud.githubusercontent.com/assets/2587901/16035459/e1cf97b4-321f-11e6-8cd6-32b29ab5b2de.png)
#### Expected behavior:

If product variants are directly accessible in the sales view, they should be directly accessible from the inventory view as well.

In #8254 it is mentioned that product variants are accessible via the Warehouse view's Configuration section. However this no longer seems to be the case in 9.0.

![image](https://cloud.githubusercontent.com/assets/2587901/16035627/c2fa13ea-3220-11e6-93ca-2fbe3e95d03f.png)

Besides, I agree with @fclementic2c: it makes little sense to me to hide the variants in the configuration in the Warehouse view: why are Sales users by default more interested in variants?
#### Discussion

There may be some configurations where you want variants to be visible to sales, but not to the warehouse. However, I think it is equally likely that if sales people deal with product variants, the warehouse wants and needs to deal with them as well. Since stock is different for different variants of the same product, the basic Product view can in fact be misleading, since it combines the stocks of all the variants.

Also, I have done some testing, and it is perfectly possible for a warehouse user to view variants. They need to go to the form of some product that has variants, open the list view of its variants, and then remove the filter that limits the variants shown to only the selected product. So there are no access right issues, but I think we can agree that this is a somewhat complicated way to get to the view.

So in summary:
- It seems to me there are no access right problems with an Inventory user accessing the tree view of product variants (`product.product`)
- I do not think that the menu should be in the same place as it is in 8.0: I think it should be in the same menu as the Products menu item is now: Inventory -> Inventory control

This hiding of product variants seems to have been done at least since 8.0: I welcome a discussion about why this was done. Perhaps there are subtle reasons why the menu is not visible to the warehouse in the same way. Would maybe @Yenthe666 know if anyone has strong feelings about this?

**Comment by Yenthe666** [comment]
Hi @aisopuro,

Thanks a lot for the detailed report, nicely written.
I for one would be for this idea, I personally think that a new menuitem could be useful here and the person that doesn't need it doesn't use it, it is as simple as that.
I however am not the one to make this decision, so I can only throw in my opinion.
@mart-e what do you think about this?

**Comment by Yenthe666** [comment]
@jco-odoo could you give a final call here? Would we add a menuitem for product variants in the inventory module or not?

**Comment by sle-odoo** [comment]
Hello,
This is solved in v11. A menu item for the variants is available in inventory as well as in sales.

Regards
